lib/idle_session_expiration_store.rb

```diff
@@ -0,0 +1,15 @@
+require 'redis-session-store'
+
+class IdleSessionExpirationStore < RedisSessionStore
+  def get_session(env, sid)
+    super.tap { update_session_ttl(sid) }
+  end
+
+  def update_session_ttl(sid)
+    expiry = default_options[:expire_after]
+    redis.expire(prefixed(sid), expiry)
+  rescue Errno::ECONNREFUSED, Redis::CannotConnectError => error
+    on_redis_down.call(error, env, sid) if on_redis_down
+    return false
+  end
+end
```

Comment:
Looks like the gem works as expected out of the box. Here's how I tested:
1. I made the session store look like this:
```ruby
options = {
  key: '_upaya_session',
  redis: {
    driver: :hiredis,
    expire_after: Figaro.env.session_timeout_in_minutes.to_i.minutes,
    key_prefix: "#{Figaro.env.domain_name}:session:",
    url: Figaro.env.redis_url
  },
  # on_redis_down: ->(e, env, sid) { do_something_will_ya!(e) },
  # on_session_load_error: ->(e, sid) { do_something_will_ya!(e) },
  serializer: :marshal
}
Rails.application.config.session_store :redis_session_store, options
```
2. `redis-cli flushall`
3. `bin/setup`
4. Set `Figaro.session_timeout_in_minutes` to `1`
5. `foreman start`
6. Visit localhost:3000
7. Open a Redis console with `redis-cli`, select the "0" database, `select 0`, and look at all the keys: `keys *`. 
8.  Verify there is a key that starts with `localhost:3000:session`.
9. Wait 1 minute
10. run `keys *` again and verify the key is gone

Also tested signing in and letting session time out on its own. When the user is signed out, the previous session they had is removed from Redis, and then they immediately get a new session. If they wait another minute, that session gets removed as well and no new sessions will appear in Redis until a new request is made.